lib/plugins/aws/package/compile/events/apiGateway/lib/hack/updateStage.js

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable no-use-before-define */
+
+'use strict';
+
+const _ = require('lodash');
+const BbPromise = require('bluebird');
+
+// NOTE --> Keep this file in sync with ../stage.js
+
+// NOTE: This code was written since there are problem setting up dedicated CloudFormation
+// Stage resource (see https://github.com/serverless/serverless/pull/5692#issuecomment-467849311 for more information).
+
+module.exports = {
+  updateStage() {
+    // this array is used to gather all the patch operations we need to
+    // perform on the stage
+    this.apiGatewayStagePatchOperations = [];
+    this.apiGatewayStageState = null;
+
+    return BbPromise.bind(this)
+      .then(getAccountId)
+      .then(getRestApiId)
+      .then(getStage)
+      .then(handleTracing)
+      .then(handleLogs)
+      .then(handleTags)
+      .then(applyUpdates);
+  },
+};
+
+function getAccountId() {
+  return this.provider.getAccountId().then((id) => this.accountId = id); // eslint-disable-line
```

Comment:
Just wondering: Is there a possible scenario where we call `this.provider.getAccountId` more than once?
If so, if I read clearly we repeat same http requests for no valid reason.

(of course it's not a concern of this PR, but maybe a thing to put somewhere on a list for improve)

Reply:
🤔 I don't think so. But a good point nontheless. Maybe we should capture that thought somewhere (as you proposed).

Reply:
I'll have an eye on it, it may require some generic approach. I'll possibly get back to when I'll have better familiarity with project as a whole.